app/controllers/api/foods_controller.rb

```diff
@@ -1,13 +1,59 @@
 class API::FoodsController < ApplicationController
   # GET /foods
+  # Consider auth tokens for API access
+  before_action :authenticate_user_from_token!, except: [:index, :show]
+
   def index
     @foods = Food.for_sale.all
     render json: @foods
   end
 
+  # POST /foods
+  def create
+    @food = Food.new(food_params)
+
+    if @food.save
+      render json: { success: true, message: "Food was successfully created.", id: @food.id, url: food_path(@food) }
+    else
+      render json: { success: false, message: "Error" }
+    end
+  end
+
+  # PATCH/PUT /foods/1
+  def update
+    if @food.update(food_params)
+      render json: { success: true, message: "Food was successfully updated.", id: @food.id, url: food_path(@food) }
+    else
+      render json: { success: false, message: "Error" }
+    end
+  end
+
   # GET /foods/1
   def show
     @food = Food.find(params[:id])
     render json: @food
   end
+
+  # DELETE /foods/1
+  # FIXME make this only depend
+  def destroy
+    @food = Food.find(params[:id])
+    if @food.seller == current_user
+      @food.destroy
+      render json: { success: true, notice: 'Food was successfully destroyed.' }
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.